src/Control/Lens/Internal/TH.hs

```diff
@@ -216,7 +212,6 @@ inlinePragma :: Name -> [DecQ]
 
 #ifdef INLINING
 
-#if MIN_VERSION_template_haskell(2,8,0)
 
 # ifdef OLD_INLINE_PRAGMAS
 -- 7.6rc1?
```

Comment:
This comment (and similar comments below) suggest that this is only used on circa-7.6 versions of GHC, which makes me wonder if this code can be trimmed down even further.

Reply:
`OLD_INLINE_PRAGMAS` is toggable by flags. I haven't looked closely what they are about.

Reply:
This arises from here:

https://github.com/ekmett/lens/blob/c073e468b67683c2a92adc5c80251b840612b8e3/lens.cabal#L343-L344

Which, in turn, has this description:

https://github.com/ekmett/lens/blob/c073e468b67683c2a92adc5c80251b840612b8e3/lens.cabal#L155-L158

Based on this, this appears to be a hack to support a release candidate version of GHC. (And an ancient one at that!) Yuck. I feel pretty confident that we can just rip out this hack now.

Speaking of which, there's several more unnecessary `impl`s in the `.cabal` file. I found the following:

* https://github.com/ekmett/lens/blob/c073e468b67683c2a92adc5c80251b840612b8e3/lens.cabal#L241-L243
* https://github.com/ekmett/lens/blob/c073e468b67683c2a92adc5c80251b840612b8e3/lens.cabal#L339-L341
* https://github.com/ekmett/lens/blob/c073e468b67683c2a92adc5c80251b840612b8e3/lens.cabal#L349-L350
* https://github.com/ekmett/lens/blob/c073e468b67683c2a92adc5c80251b840612b8e3/lens.cabal#L359-L360

Let's drop these `impl`s as well.